### ISSUE TITLE
Stateless components cannot be arrow functions

### ISSUE THREAD

**Issue by LinusU** [developit/preact#122]
![screen shot 2016-04-12 at 11 00 33](https://cloud.githubusercontent.com/assets/189580/14455053/cd092a2e-009d-11e6-9e03-bcfa724db812.png)

A bug in `isFunctionalComponent` is causing an error to be thrown when passing an arrow function as a stateless component. This is because arrow functions doesn't have a prototype, which is used in this function.

Compare these two identity functions:

![screen shot 2016-04-12 at 11 02 39](https://cloud.githubusercontent.com/assets/189580/14455110/1c9441b4-009e-11e6-9969-65c356e6a143.png)

This is the very simple app that I'm using together with browserify:

```javascript
const { h, render } = require('preact')

const App = (props) =>
  h('h1', null, 'Hello World')

render(h(App), document.body)
```

Note that I'm not using babel here. This works when using babel because it converts the arrow functions to normal functions.

**Comment by developit** [comment]
Ah, very good catch. Should be a quick fix.

**Comment by LinusU** [comment]
Thanks for the quick fix, if you could publish a patch release right away I would be _so_ happy! 🎉

**Comment by developit** [comment]
@LinusU yes sorry - meant to publish but I had to step out for a sec. Will publish in ~20 minutes. Cheers!

**Comment by developit** [comment]
By the way, not sure if it helps but I've been publishing alphas here: https://gist.github.com/developit/75e724607fe33de9e63f
For larger changes I generally publish a `beta` dist-tag to NPM.

**Comment by LinusU** [comment]
Nice! 🙌 

I'm using npm so it might be a bit tedious to include the gist...

**Comment by developit** [comment]
For sure, yup.

**Comment by developit** [comment]
Released as [4.6.1](https://github.com/developit/preact/releases/tag/4.6.1). Thanks!

**Comment by LinusU** [comment]
Thank you!
